aimmo-game/simulation/world_map.py

```diff
@@ -58,13 +57,28 @@ def __hash__(self):
         return hash(self.location)
 
     @property
-    def moves(self):
-        return [move for move in self.actions if isinstance(move, MoveAction)]
+    def location(self):
+        return self._location
+
+    @property
+    def is_habitable(self):
+        return self._habitable
 
     @property
     def is_occupied(self):
         return self.avatar is not None
 
+    @property
+    def avatar(self):
+        return self._avatar
+
+    @avatar.setter
+    def avatar(self, new_avatar):
+        if new_avatar is not None:
+            assert self.is_habitable, 'Cannot move avatar to inhabitable cell.'
+            assert self._avatar is None, 'Cannot move avatar to occupied cell.'
```

Comment:
Checks and validation in property setters are fine imo (it's basically what they're for). And I think asserts are fine for conditions that should *never* happen like these.